tests/backend_test/main.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -euo pipefail
+
+#. ../assert.sh continue_on_error #  load test functions with continue_on_error/stop_on_error
+
+#Cleanup
+terraform init -backend-config=./test_backend.tfvars test3
+terraform destroy -auto-approve test3
+
+#Check if resource group exists
+#Resource group does not exists
+assertRG core-usea-rg-test not
+
+#Create Resource group and Keyvault 1
+terraform init -backend-config=./test_backend.tfvars test1
+terraform apply -auto-approve test1
+
+#Check if resource group exists
+assertRG core-usea-rg-test
+assertResource core-usea-rg-test "Microsoft.KeyVault/vaults" core-usea-kv1-test
+
+#Key Vault 2 does not exists
+assertResource core-usea-rg-test "Microsoft.KeyVault/vaults" core-usea-kv2-test not
+
+terraform init -backend-config=./test_backend.tfvars test2
+#Deploy Key Vault 2 only
+terraform apply -auto-approve test2
+
+assertRG core-usea-rg-test
+assertResource core-usea-rg-test "Microsoft.KeyVault/vaults" core-usea-kv1-test
+assertResource core-usea-rg-test "Microsoft.KeyVault/vaults" core-usea-kv2-test
+before=`az resource list -g core-usea-rg-test --resource-type "Microsoft.KeyVault/vaults"|jq .[].name| wc -l`
```

Comment:
The use of `jq `here is introducing a dependency that we don't need.  Please use the built-in `--query` feature of Azure CLI to query the results.  Reach out to me if you need any help constructing the query.